viewing_party/party.py

```diff
@@ -1,23 +1,139 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    if title and genre and rating:
+        return {
+        "title": title,
+        "genre": genre, 
+        "rating": rating
+        }
+    else:
+        return None
+        
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data 
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    # watchlist = user_data["watchlist"]
+    # watched = user_data["watched"]
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
+
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    list_of_ratings = []
+    for movie in user_data["watched"]:
+        list_of_ratings.append(movie["rating"])
+    if len(user_data["watched"]) == 0:
+        avg_rating = 0

```

Comment:
Consider doing the list length check first and reversing the comparison to look for the empty list, using the check as a guard clause. This would allow us to avoid indenting the main logic of the function. We can also check whether a list is empty slightly more pythonically

```py
    if not user_data['watched']:
        return 0.0

    # main logic for calculating the average (unindented)
```